Create UserProfile when User first logs in

mojcas opened the pull request:
Create UserProfile when User first logs in; change model to default activity status=KITTEN (no LAZY cats here :) )
Addresses issue #216

anuschka commented:
:+1:

ialja commented on api/models.py:
```diff
@@ -14,7 +16,7 @@ class UserProfile(models.Model):
 	website = models.URLField(blank=True)
 	bio = models.TextField(max_length=2000, blank=True)
-	activity_status = models.CharField(max_length=50, choices=ACTIVITY_LEVEL_CHOICES, default='ACTIVE')
+	activity_status = models.CharField(max_length=50, choices=ACTIVITY_LEVEL_CHOICES, default='KITTEN')
```
Re: default=kitten - what about somebody, who signs up and is not a member of CodeCatz?

mojcas replied:
I would add some more logic here, to ask -- do you want to become a member of CodeCatz. If they say yes, then they would be assigned KITTEN activity status, and they can't change it themselves. If they say no, then nothing would change for them (sign in but no authority to do anything). Would that be okay?

ialja commented on api/models.py:
```diff
@@ -2,11 +2,13 @@
-	('LAZY', 'Lazy Cat'),
+	('KITTEN', 'New Kitten'),
 	('ACTIVE', 'Active Cat'),
+	('SLEEPY', 'Sleepy Cat'),
```
Changing models too often is generally not a good idea. Before changing something like this, always check whether the current names are used anywhere else on the site (hint: they are) and whether you're breaking anything with your new features..

mojcas replied:
I've checked, and LAZY is not used anywhere, but on member.html template

mojcas commented:
When user logs in, user object is created, but no user profile exists yet, so member page doesn't have anything to display. I guess the question is, do we want user profile be created withI new user, or not? Why would anyone sign in, and not be a member? I though assigning a special kitten default activity level for new users would be good.. Maybe I am missing something.

ialja commented:
It's a good idea to create a user profile object for each new user. Once TrackCat goes public, **ANYONE** in the world (incl. spammers and trolls) will be able to click the "Sign in with GitHub", so I'm not sure you want that anyone to be automatically be displayed on the page. 

"Why would anyone sign in, and not be a member?" - because they can. And if they have rights to edit stuff, they will. Because they can. It's your job to prevent random users from making a mess out of your website.

mojcas commented:
I have removed displaying KITTEN from Members page, and removed activity status from form so members can not change it themselves, but can still edit and update their profile..

mojcas commented:
Another thing I found, is to check for permissions of the user.. Only users with permissions can add new project, task, or change them...

livike commented:
The new create/add/delete permission change is cool and works fine.

Also the Userprofile creation is working, but maybe after the first login would be nice to redirect to some welcome message, to prom them to fill in their profile or a message that a staff member will review their membership...

ialja commented:
@livike Do add that suggestion to Issues!

gandalfar commented:
Thanks for review @livike, merging
